app/models/health/careplan_saver.rb

```diff
@@ -22,57 +20,86 @@ def create
       end
     end
 
-
     def update
       success = true
-      # limited to only signatures 11/27 per request from BHCHP, only save QA for signatures
-      should_really_create_qa = @careplan.just_signed? && @create_qa
       @careplan.compact_future_issues
+
+      # limited to only signatures 11/27 per request from BHCHP, only save QA for signatures
       begin
         @careplan.class.transaction do
-          if should_really_create_qa
-            signature_date = @careplan.provider_signed_on
-            @qualifying_activity.date_of_activity = signature_date
-
-            case @careplan.provider_signature_mode.to_s
-            when 'email'
-              @qualifying_activity.mode_of_contact = :other
-              @qualifying_activity.reached_client = :collateral
-              @qualifying_activity.mode_of_contact_other = 'On-line'
-              @qualifying_activity.reached_client_collateral_contact = 'On-line Signature'
-            when 'in_person'
-              @qualifying_activity.mode_of_contact = :in_person
-              @qualifying_activity.reached_client = :yes
-            else
-              # default to email signature
-              @qualifying_activity.mode_of_contact = :other
-              @qualifying_activity.reached_client = :collateral
-              @qualifying_activity.mode_of_contact_other = 'On-line'
-              @qualifying_activity.reached_client_collateral_contact = 'On-line Signature'
-            end
+          if @careplan.just_finished? && @create_qa
+            qualifying_activity = setup_care_planning_qualifying_activity
```

Comment:
If both signatures are collected in one go (added manually, for instance) do we need to generate both QA?

Reply:
Good catch! Fixed.